Resources/Prototypes/Entities/Structures/Power/Generation/Tesla/energyball.yml

```diff
@@ -0,0 +1,121 @@
+- type: entity
+  id: TeslaEnergyBall
+  name: ball lightning
+  description: A giant ball of pure energy. The space around it is humming and melting.
+  components: 
+  - type: Clickable
+  - type: AmbientSound
+    volume: 2
+    range: 14
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+  - type: CanMoveInAir
+  - type: Fixtures
+    fixtures:
+      TeslaCollider:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.35
+        hard: true
+        restitution: 0.8
+        density: 99999
+        mask:
+        - Opaque  
+        layer:
+        - GlassLayer  
+  - type: TeslaEnergyBall
+    energyLoss: 3
+    spawnProto: TeslaMiniEnergyBall 
+    soundCollapse:
+      path: /Audio/Effects/tesla_collapse.ogg
+      params:
+        variation: 0.3
+  - type: LightningArcShooter
+    arcDepth: 3
+    maxLightningArc: 4
+    shootRange: 7
+    shootMinInterval: 2
+    shootMaxInterval: 5
+    lightningPrototype: HyperchargedLightning
+  - type: ChasingWalk
+    maxSpeed: 2.5
+    minSpeed: 1.875
+    chasingComponent: LightningTarget
+  - type: ChaoticJump
+    jumpMinInterval: 8
+    jumpMaxInterval: 15
+    offset: 1
+  - type: PointLight
+    enabled: true
+    radius: 5
+  - type: Appearance
+  - type: WarpPoint
+    follow: true
+    location: tesla ball
+  - type: Sprite
+    drawdepth: Effects
+    sprite: Structures/Power/Generation/Tesla/energy_ball.rsi
+    shader: unshaded
+    layers:
+    - state: energy_ball
+  - type: EmitSoundOnSpawn
+    sound: 
+      path: /Audio/Effects/tesla_collapse.ogg
+      params:
+        variation: 0.3
+    
+- type: entity 
+  id: TeslaMiniEnergyBall
+  name: mini ball lightning
+  description: The cub of a destructive energy cage. Not as dangerous, but still not worth touching with bare hands.
+  components: 
+  - type: Clickable
+  - type: AmbientSound
+    volume: -2
+    range: 6
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+  - type: CanMoveInAir
+  - type: Fixtures
+    fixtures:
+      TeslaCollider:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.35
+        hard: true
+        restitution: 0.8
+        density: 99999
```

Comment:
Found collision problems: Tesla eats and charges from its lightning bolts. Small lightning bolts have a dilemma: they must have a collision to repel from the containment field. They must have 99999 density so that touching the field doesn't accelerate them to first space speed. But given these parameters, they fly and push people and objects apart :trollface: